lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -2005,8 +2015,30 @@ defaultSettings = Settings {
     poolMetadataSource = FetchNone
 }
 
+-- | Various internal states of the pool DB
+--  that need to survive wallet restarts. These aren't
+--  exposed settings.
+{-# HLINT ignore InternalState "Use newtype instead of data" #-}
+data InternalState = InternalState
+    { lastMetadataGC :: POSIXTime
+    } deriving (Generic, Show, Eq)
+
+defaultInternalState :: InternalState
+defaultInternalState = InternalState
+    { lastMetadataGC = fromIntegral @Int 0 }
```

Comment:
**Question**: does this mean the default value of `lastMetadataGC` is `1970-01-01 00:00 UTC`? Hopefully, we wouldn't pass this value back to API callers.

**Suggestion**: Use `Maybe POSIXTime` here to indicate the possibility that there hasn't been a garbage collection yet.

```hs
data InternalState = InternalState
    { lastMetadataGC :: Maybe POSIXTime
    } deriving (Generic, Show, Eq)


```

Reply:
Yes it does. Using Maybe in JSON results in a nullable type. That's ambiguous. It doesn't say there was no GC. We'd have to wrap it in an actual sum type. I don't think there's any value in this.

Reply:
We already represent the `lastGC` value as a `Maybe` in the API:

```hs
data ApiListStakePools apiPool = ApiListStakePools                                                                                                                                    
    { pools :: [apiPool]                                                                                                                                                              
    , lastGC :: !(Maybe (ApiT Iso8601Time))                                                                                                                                           
    } deriving (Eq, Generic, Show)  
```

IMO, `Nothing` is a cleaner representation than returning a magic value of `1970-01-01 00:00 UTC`, especially given that we are already using `Maybe` in the API.

Reply:
Yes, the `Maybe` here is because of Jormungandr, which doesn't do GC at all. Shelley can always expect a value. Our types don't allow to express this.